rmw_fastrtps_shared_cpp/src/rmw_service.cpp

```diff
@@ -102,12 +103,6 @@ __rmw_destroy_service(
       show_previous_error();
       RMW_SET_ERROR_MSG("Fail in delete datareader");
       final_ret = RMW_RET_ERROR;
-      info->request_reader_->set_listener(nullptr);
```

Comment:
Do not remove this line. In case that the DataReader deletion fails, the listener is detached from the DataReader so it can be deleted without having any conflict. Same comment applies below to the `response_writer`.

Reply:
`response_writer` listener is not set to `nullptr` in case that the deletion of the DataWriter fails. This could cause a core dump if the deleted listener is still attached to the DataWriter.